Replace tipsy with tooltip

irgendwie opened the pull request:
As we moved from tipsy to tooltip (https://github.com/owncloud/core/pull/17075), I changed them for the mail app.

This resolves some issues / problems we had, like the recently https://github.com/owncloud/mail/issues/929, but also creates new problems:
- [x] weird transparency problems (the 'mail to' tooltips) @jancborchardt ideas?

Please review this and search for bugs! :+1:

jancborchardt commented:
I’m pretty sure the transparency problems come from using opacity on the element. The tooltip is in the container of the element, so instead of opacity we need to use text-color with alpha.

irgendwie commented:
@jancborchardt Awesome! Changed the container of the tooltip to body. :+1:
Should be ok now.

jancborchardt commented:
Ok now maybe squash your commits into one, to keep the history clean. :)

irgendwie commented:
Pfff...

jancborchardt commented:
:D no pff here! Keep stuff clean my friend

irgendwie commented:
done :stuck_out_tongue_winking_eye:

itheiss commented:
Ok. Here we go ....

1. The tooltip looks a bit odd. But I assume this is beause I am testing against 8.1 and Bootstrap tooltip is only available in 8.2, right? So this is no issue.

 ![tooltip](https://cloud.githubusercontent.com/assets/2655169/9268182/fccefa78-425b-11e5-9382-6bbe74c28cc5.jpg)

2. There is no tooltip when I move the mouse over the trash icon. But this was also an issue before (The trash icon to delete an account shows a tooltip for me?!?)

Although I can't reproduce #923 everything is working as expected :+1:

jancborchardt commented on js/views/composer.js:
```diff
@@ -124,6 +124,8 @@ views.Composer = Backbone.View.extend({
 			replyCcList: options.data.replyCcList
 		});
+		
+		$('.tipsy-mailto').tooltip({placement:'bottom', container:'body'});
```
While at it, can you rename the .tipsy-mailto class to `.mailto-tooltip`?

jancborchardt commented:
I would say to fix this and make it still compatible with ownCloud stable8.1 and older, there should be a check around the .tooltip call if it exists. If it does not exist, use .tipsy. @irgendwie Ok?

@Henni @PVince81 @MorrisJobke we also need to keep this in mind for other apps.

PVince81 commented:
Well, or third party apps can simply import tipsy if they need it instead of relying on core bundling it.
Ideally apps should not rely too strongly on libraries shipped in core.
Else this would qualify as "breaking the API".

Another alternative is providing a tispy shim in core to ease the transition.

DeepDiver1975 commented:
:-1: but this change will break the app on anything less then 8.2

There was the idea that only the implementation of the tipsy/tooltip foo changes in core - the "API" should stay the same - otherwise all apps break

DeepDiver1975 commented:
> Another alternative is providing a tispy shim in core to ease the transition.

That shim was a requirement for me - not in there? :imp:

DeepDiver1975 commented:
okay - here is the code I was looking for: https://github.com/owncloud/core/blob/56cff46dc32a0049840bb09f1769413eace064d6/core/js/js.js#L1846

No need to change anything in the mail app from my pov

Henni commented:
There are some problems with the shim as bootstrap tooltips don't support the live setting for example.

DeepDiver1975 commented:
Then we pull thebootstrap tooltip into the app

jancborchardt commented:
@irgendwie any updates? :)

jancborchardt commented:
We should also fix this for 0.2 since the tooltips are broken. @irgendwie can you just copy the library from core into the app so it works on older ownCloud versions too?

irgendwie commented:
@jancborchardt Just copying the library would break the tooltips, because I would also need to import the stylesheet...

I now changed the position of tipsy creation, therefore they should work with both, using tipsy and the tooltip shim.
Also fixed #929 with the last commit.

Please test and give feedback!

jancborchardt commented:
Is it just that the tooltips are slightly transparent? I would say that is ok.

irgendwie commented:
@jancborchardt It's because of the container of the tooltip. With tipsy, there is no problem with the transparency. As I can't control the container while using the tooltip/tipsy shim, the transparency problem is back.

jancborchardt commented:
@irgendwie so apart from the minor transparency stuff this is now good to go?

ChristophWurst commented:
:+1: works!

jancborchardt commented:
Nice! :) Now onwards with the last stuff for 0.2
